protocol/xmrmaker/claim.go

```diff
@@ -99,6 +108,53 @@ func (s *swapState) claimFunds() (ethcommon.Hash, error) {
 	return txHash, nil
 }
 
+// discoverRelayersAndClaim discovers available relayers on the network,
+func (s *swapState) discoverRelayersAndClaim() (ethcommon.Hash, error) {
+	relayers, err := s.Backend.DiscoverRelayers()
+	if err != nil {
+		return ethcommon.Hash{}, err
+	}
+
+	forwarderAddress, err := s.Contract().TrustedForwarder(&bind.CallOpts{})
```

Comment:
`&bind.CallOpts{Context: s.ctx}`  (otherwise you could just pass `nil` and get the same result).